SignalMessaging/environment/AppSetup.m

```diff
@@ -40,6 +41,10 @@ + (void)setupEnvironment:(CallMessageHandlerBlock)callMessageHandlerBlock
                                                   profileManager:OWSProfileManager.sharedManager];
         [TextSecureKitEnv setSharedEnv:sharedEnv];
 
+        // Register renamed classes.
+        [NSKeyedUnarchiver setClass:[OWSUserProfile class] forClassName:[OWSUserProfile collection]];
+        [NSKeyedUnarchiver setClass:[OWSDatabaseMigration class] forClassName:[OWSDatabaseMigration collection]];
```

Comment:
Not strictly necessary, but I think it'll be convenient to ensure `NSKeyedUnarchiver` knows how to load any entity in YDB at all times.

Reply:
👍 

I think this makes sense - but to make sure I understand, whenever a class stored in Yap is renamed we should register a legacy handler here?

I guess the exception being collections that have more than one object type stored in them, which off the top of my head is only the thread collection (where TSContactThread and TSGroupThread cohabit), which I don't imagine we'll be renaming any time soon.

Reply:
> whenever a class stored in Yap is renamed we should register a legacy handler here?

That's right.  We don't have to be strict about it, but it provides some safety and convenience.

> I guess the exception being collections that have more than one object type stored in them,

I could be wrong, but I think there's a distinction between the collection name and the class name.  I'm just using the `collection` property of these classes as a convenience, since the collection name and class name coincided before these classes were renamed.  

Let's consider a worst case.  Suppose we stored class A and class B in a shared collection C and we renamed A -> A2 and B -> B2, we would then:

* Make sure `A2.collection` and `B2.collection` both returned C.
* `[NSKeyedUnarchiver setClass:[A2 class] forClassName:@"A"];
* `[NSKeyedUnarchiver setClass:[B2 class] forClassName:@"B"];

Right?

Reply:
That sounds right to me.